modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImport/ModalImportContent.tsx

```diff
@@ -195,11 +212,30 @@ export function ModalImportContent({
 									try {
 										const JSONFile = JSON.parse(
 											fileReader.result as string
-										) as {externalReferenceCode: string};
-										setError(undefined);
-										setExternalReferenceCode(
-											JSONFile.externalReferenceCode
-										);
+										) as
+											| {externalReferenceCode: string}
+											| ObjectDefinition[];
+
+										if (
+											Liferay.FeatureFlags[
+												'LPS-187142'
+											] &&
+											Array.isArray(JSONFile) &&
+											JSONFile[0].scope
+										) {
+											setError(undefined);
+											setObjectDefinitions(JSONFile);
+											setExternalReferenceCode('');
+										}
```

Comment:
Hey @danvb112, I believe we can clean de `name` state here. Since the user can fill the name input before select the file. What do you think?

Reply:
Hey @igor-franca, sure! I'll clear the name here